codemirror/codemirror.d.ts

```diff
@@ -1125,6 +1125,127 @@ declare namespace CodeMirror {
+         * of lines to leave visible around such stretches (which defaults to 2). Defaults to false.
+         */
+        collapseIdentical?: boolean | number;
+
+        /**
+         * Sets the style used to connect changed chunks of code. By default, connectors are drawn. When this is set to "align",
+         * the smaller chunk is padded to align with the bigger chunk instead.
+         */
+        connect?: string;
+
+        /**
+         * Callback for when stretches of unchanged text are collapsed.
+         */
+        onCollapse?: MergeViewCollapser;
+
+        /**
+         * Provides original version of the document to be shown on the right of the editor.
+         */
+        orig: any;
+
+        /**
+         * Provides original version of the document to be shown on the left of the editor.
+         * To create a 2-way (as opposed to 3-way) merge view, provide only one of origLeft and origRight.
+         */
+        origLeft?: any;
+
+        /**
+         * Provides original version of document to be shown on the right of the editor.
+         * To create a 2-way (as opposed to 3-way) merge view, provide only one of origLeft and origRight.
+         */
+        origRight?: any;
+
+        /**
+         * Determines whether buttons that allow the user to revert changes are shown. Defaults to true.
+         */
+        revertButtons?: boolean;
+
+        /**
+         * When true, changed pieces of text are highlighted. Defaults to true.
+         */
+        showDifferences?: boolean;
+    }
+
+    interface MergeViewEditor extends Editor {
+        /**
+         * Returns the editor instance.
+         */
+        editor(): Editor;
+
+        /**
+         * Left side of the merge view.
+         */
+        left: DiffView;
+        leftChunks(): MergeViewDiffChunk;
+        leftOriginal(): Editor;
+
+        /**
+         * Right side of the merge view.
+         */
+        right: DiffView;
+        rightChunks(): MergeViewDiffChunk;
+        rightOriginal(): Editor;
+
+        /**
+         * Sets whether or not the merge view should show the differences between the editor views.
+         */
+        setShowDifferences(showDifferences: boolean): void;
+    }
+
+    /**
+     * Callback for when stretches of unchanged text are collapsed.
+     */
+    interface MergeViewCollapser {
+        (mergeView: MergeViewEditor, line: number, size: number, mark: TextMarker): void;
+    }
+
+    interface DiffView {
+        /**
+         * Forces the view to reload.
+         */
+        forceUpdate(): DiffViewUpdater;
+
+        /**
+         * Sets whether or not the merge view should show the differences between the editor views.
+         */
+        setShowDifferences(showDifferences: boolean): void;
+    }
+
+    /**
+     * Determines how the view should be updated. If "full" mode is given it clears everything before updating the diff.
+     */
+    interface DiffViewUpdater {
```

Comment:
Same here, can we put this declaration where it's used in-place, rather than put behind a name.
Thanks!